### DISCUSSION TITLE
I can't stablish connection to DB

### DISCUSSION THREAD

**Discussion by Bullsito** [OwlGamingCommunity/V#12] [Support]
[RAGEMP Console]

Unable to read data from the transport connection: A connection established by the software on your host computer has been aborted ...
The type initializer for 'Database.LegacyFunctions' threw an exception.
The type initializer for 'Database.LegacyFunctions' threw an exception.
The type initializer for 'Database.LegacyFunctions' threw an exception.
The type initializer for 'Database.LegacyFunctions' threw an exception.
Unhandled exception. System.TypeInitializationException: The type initializer for 'Database.LegacyFunctions' threw an exception.
 ---> System.IO.IOException: Unable to read data from the transport connection: A connection established by the software on your host computer has been aborted ...
 ---> System.Net.Sockets.SocketException (10053): A connection established by the software on your host computer has been aborted ...
   at MySqlStream MySql.Data.Common.Ssl.StartSSL(ref Stream baseStream, Encoding encoding, string connectionString)
   at void MySql.Data.MySqlClient.NativeDriver.Open()
   at void MySql.Data.MySqlClient.Driver.Open()
   at Driver MySql.Data.MySqlClient.Driver.Create(MySqlConnectionStringBuilder settings)
   at Driver MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnection()
   at Driver MySql.Data.MySqlClient.MySqlPool.GetPooledConnection()
   at Driver MySql.Data.MySqlClient.MySqlPool.TryToGetDriver()
   at Driver MySql.Data.MySqlClient.MySqlPool.GetConnection()
   at void MySql.Data.MySqlClient.MySqlConnection.Open()
   at bool Database.MySQLInstance.Initialize(bool bIsStartup)
   at new Database.MySQLInstance()
   at static Database.LegacyFunctions()
   --- End of inner exception stack trace ---
   at Database.LegacyFunctions.GetAllCustomInteriors()
   at MapLoader.Init()
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__139_1(Object state)
   at System.Threading.QueueUserWorkItemCallbackDefaultContext.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
The type initializer for 'Database.LegacyFunctions' threw an exception.

[VS Terminal]

Exception: 'System.AggregateException' at System.Private.CoreLib.dll
Exception: 'System.IO.IOException' at MySql.Data.dll
Exception: 'System.IO.IOException' at System.Net.Security.dll
Exception: 'System.IO.IOException' at System.Private.CoreLib.dll
Exception: 'System.AggregateException' at System.Private.CoreLib.dll
Bootstrapper Error: 0 : Unable to read data from the transport connection: A connection established by the software on your host computer has been aborted ...
Exception: 'System.TypeInitializationException' at owl_mysql.dll
Exception: 'System.TypeInitializationException' at System.Private.CoreLib.dll
Unhandled Exception 'System.TypeInitializationException' at System.Private.CoreLib.dll
The type initializer for 'Database.LegacyFunctions' threw an exception.

Exception: 'System.TypeInitializationException' at owl_mysql.dll

**Comment by 1Example** [comment]
Try to enable ssl on your mysql server.

**Reply by Bullsito** [reply]
I'm using MariaDB, when I'll try to use MySQL I had access errors like the password or user were incorrect (although they were right) and I can't activate SSL in MariaDB or MySQL (I'm using Windows)

**Reply by 1Example** [reply]
Look it up, you can have SSL on windows too.

**Reply by Bullsito** [reply]
I try it to enable it yesterday and I couldn't. I followed this tutorial: https://serverfault.com/questions/783861/enabling-ssl-in-mysql-when-using-windows-as-a-server-and-client

MariaDB [(none)]> show global variables like '%ssl%';
+---------------------+----------------------------------+
| Variable_name       | Value                            |
+---------------------+----------------------------------+
| have_openssl        | NO                               |
| have_ssl            | YES                              |
| ssl_ca              | ..\..\mysqlCerts\ca-cert.pem     |
| ssl_capath          |                                  |
| ssl_cert            | ..\..\mysqlCerts\server-cert.pem |
| ssl_cipher          |                                  |
| ssl_crl             |                                  |
| ssl_crlpath         |                                  |
| ssl_key             | ..\..\mysqlCerts\server-key.pem  |
| version_ssl_library | WolfSSL 4.8.0                    |

----

MariaDB [(none)]> status

C:\Program Files\MariaDB 10.6\bin\mysql.exe  Ver 15.1 Distrib 10.6.4-MariaDB, for Win64 (AMD64), source revision 2db692f5b4d6bb31a331dab44544171c455f6aca

Connection id:          3
Current database:
Current user:           root@localhost
SSL:                    Not in use
Using delimiter:        ;
Server:                 MariaDB
Server version:         10.6.4-MariaDB mariadb.org binary distribution
Protocol version:       10
Connection:             localhost via TCP/IP
Server characterset:    utf8mb3
Db     characterset:    utf8mb3
Client characterset:    cp850
Conn.  characterset:    cp850
TCP port:               3306
Uptime:                 46 sec


Should I move the DB to a local virtual Linux?

(And yes, I have openSSL installed in my windows)

EDIT:

Even If I tried to launch it like this I have this error:

Authentication to host '127.0.0.1' for user 'root' using method 'mysql_native_password' failed with message: Access denied for user 'root'@'localhost' (using password: YES)
[ERROR] MySQL Connection Failed. Cannot Connect to Server.

**Reply by 1Example** [reply]
Try to use a different authentication method for the user root and I also recommend you to create a new mysql user. Try to search for the error on google and you will find tons of solutions.

**Reply by Bullsito** [reply]
I try it with the 'ssluser' but I have some errors about the certs

**Reply by 1Example** [reply]
I'm pretty sure you will fix it by spending more time on fixing the errors you encounter. The main answer to your problem is to fix the mysql server.

**Reply by Bullsito** [reply]
There is no way to disable the SSL from the gamemode? It's like its hardcode somewhere, i can't disable it from the connection string

I'm guessing the main problem it's they are making me to connect to the DB using SSL.

**Reply by Bullsito** [reply]
So I could install the SSL Connection in **MySQL** (not MariaDB, it was impossible even in Linux, I coudn't make functional the SSL connection, if you know any guide, please share it) and make it functional but when I tried to connect my RAGE Server I have again this exception:

Authentication to host '127.0.0.1' for user 'root' using method 'caching_sha2_password' failed with message: Access denied for user 'root'@'localhost' (using password: YES)
[ERROR] MySQL Connection Failed. Cannot Connect to Server.

----

These are my parameters (changed in both files)

string[] gameSettings = new string[] {
				SettingHelpers.GetDevEnvironmentSetting("GAME_DATABASE_IP") ?? "127.0.0.1",
				SettingHelpers.GetDevEnvironmentSetting("GAME_DATABASE_NAME") ?? "gtav",
				SettingHelpers.GetDevEnvironmentSetting("GAME_DATABASE_USERNAME") ?? "ssluser",
				SettingHelpers.GetDevEnvironmentSetting("GAME_DATABASE_PASSWORD") ?? "sslpassword",
				SettingHelpers.GetDevEnvironmentSetting("GAME_DATABASE_PORT") ?? "3306"
				};


string[] authSettings = new string[] {
				SettingHelpers.GetDevEnvironmentSetting("AUTH_DATABASE_IP") ?? "127.0.0.1",
				SettingHelpers.GetDevEnvironmentSetting("AUTH_DATABASE_NAME") ?? "core",
				SettingHelpers.GetDevEnvironmentSetting("AUTH_DATABASE_USERNAME") ?? "ssluser",
				SettingHelpers.GetDevEnvironmentSetting("AUTH_DATABASE_PASSWORD") ?? "sslpassword",
				SettingHelpers.GetDevEnvironmentSetting("AUTH_DATABASE_PORT") ?? "3306"
				};

**Reply by Bullsito** [reply]
So, I manage to install and use SSL in PHPMyAdmin (finally!) but I'm still having the error:

Authentication to host '127.0.0.1' for user 'root' using method 'mysql_native_password' failed with message: Access denied for user 'root'@'localhost' (using password: YES)
[ERROR] MySQL Connection Failed. Cannot Connect to Server.
